packages/header-change-detection/lib/lambda/header-check.ts

```diff
@@ -0,0 +1,300 @@
+import axios from "axios";
+import { DynamoDBClient, BatchGetItemCommand, BatchGetItemCommandInput, KeysAndAttributes, UpdateItemCommandInput, UpdateItemCommand, AttributeValueUpdate } from "@aws-sdk/client-dynamodb";
+import { PublishCommand, PublishInput, SNSClient } from "@aws-sdk/client-sns";
+
+const URLS = process.env.URLS;
+const HEADERS = process.env.HEADERS;
+const TABLE = process.env.TABLE!;
+
+const config = ""
+const DB_CLIENT = new DynamoDBClient(config);
+
+const securityHeaders = HEADERS?.split(",") || []
+
+type Headers = Map<string, string | undefined>
+type URLHeaders = Map<string, Headers>
+
+export const handler = async () => {
+  const urls = URLS?.split(',') || [];
+
+  // Fetch stored headers
+  const [storedUrlHeaders, currentUrlHeaders] = await Promise.all([
+    getStoredValues(urls),
+    fetchHeaders(urls)
+  ]);
+
+  // Find any differences between the headers
+  const headerDifferences = new Map() as Map<string, Difference[]>;
+  let differencesDetected = false;
+  const dbUpdates = urls.map(url => {
+    const currentHeaders = currentUrlHeaders.get(url)
+    const storedHeaders = storedUrlHeaders.get(url) || new Map()
+
+    if (!currentHeaders) throw new Error(`Could not get current headers for ${url}`)
+
+    // Check all headers that we care about
+    headerDifferences.set(url, compareHeaders(securityHeaders, storedHeaders, currentHeaders));
+
+    const headersToUpdate: Headers = new Map();
```

Comment:
Similar to the above, does `new Headers()` work here? More a curiosity than anything else :D